proposals/2020/NMEC Implementation.md

```diff
@@ -0,0 +1,193 @@
+
+Metrics to be able to capture:
+- r-squared
+- Adjusted r-squared
+- RMSE
+- CVRMSE
+- Net determination bias error (NDBE)
+- MBE
+- NMBE
+
+Model inputs to capture:
+- Baseline period start datetime
+- Baseline period end datetime
+- Reporting period start datetime
+- Reporting period end datetime
+- Data interval: [Hourly, daily, monthly]
+- Confidence interval?
+- Model Type: One of above models
+- Dependent variable:
+    - ResourceUse (electricity, natural gas, etc.)
+    - Units (kWh, therms, etc.)
+Independent variables:
+    - For CPMs and SLR model, temperature is only independent variable
+    - Units
+    
+Model outputs to capture:
+- BAE
+- RAE
+- RCE
+- BCESC
+- RCESC
+- Avoided energy use
+- Normalized savings
+- Number of parameters (p)
+- Number of data points or periods in the baseline period (n)
+- Number of data points or periods in the reporting period (m)
+- Degrees of freedom (n - p)
+- Savings percentage: Believe this is just `Avoided energy use / RCE` #TODO
+- Savings uncertainty: Is this what is defined by equations 4-6 and 4-7 in [1]? Is this the same across CalTrack methods / CPUC NMEC rulebook? #TODO
+
+Potentially?
+- Cutoff thresholds (for CVRMSE or NDBE)
+- 
+
+### Approach
+- Two `auc:Scenario`s will be used, one to represent the baseline period and one to represent the reporting period.  For situations where multiple reporting periods are considered (i.e. in the case where an energy simulation is used to produce synthetic data for different potential retrofit scenarios), additional reporting period scenarios can be defined.
+- The concept of an `auc:DerivedModel` is introduced.  It represents the generic concept of either a baseline or reporting model (referred to as an adjustment model in [3]).
+- The `auc:DerivedModel` complex type will be able to be instantiated as a child of the following elements:
+    - `auc:Scenario/auc:ScenarioType/auc:CurrentBuilding`
+        - It is expected that models for baseline periods would be embedded here (regardless if modeled or measured)
+        - `auc:DerivedModelPerformance` and child elements should be captured here
+        - All elements starting with `auc:BaselinePeriod` should be captured here
+    - `auc:Scenario/auc:ScenarioType/auc:PackageOfMeasures`
+        - It is expected that models for reporting periods would be embedded here (regardless if modeled or measured)
+        - All elements starting with `auc:ReportingPeriod` should be captured here
+- Multiple `auc:DerivdeModels` should be able to be captured, as users may want to test different `auc:ModelType`s
+- The `auc:DerivedModel` will be a complex type that would look something like:
+```xml
+  <auc:DerivedModels>
+    <auc:DerivedModel ID="Model1">
+      <auc:DerivedModelPeriod>[Baseline, Reporting]</auc:DerivedModelPeriod>
+      <auc:DerivedModelBaselineID/>
+      <auc:DerivedModelInputs>
+        <auc:ModelType>[2P SLR, 3P heating CPM, 3P cooling CPM, 4P CPM, 5P CPM]</auc:ModelType>
```

Comment:
I'm wondering how much documentation we should provide on the acronyms here. I'm fine with them, but I have been working with building data for a while.

Reply:
I think this is better with the updated proposal.  Let me know.